docs/docs/comet-core-development/creating-customizable-admin-components.md

```diff
@@ -0,0 +1,239 @@
+        root: "div";
+        title: typeof Typography;
+    }> {
+    variant?: "primary" | "secondary";
+    children?: React.ReactNode;
+}
+```
+
+## Using props
+
+Props should not be used directly but through the `useThemeProps` hook instead.
+This ensures that the components `defaultProps` from the theme are merged with the passed-in props.
+
+Note that the name passed to the `useThemeProps` hook must be the same as the component's name, prefixed with `CometAdmin`.
+
+```tsx
+export function MyComponent(inProps: MyComponentProps) {
+    const { children, variant } = useThemeProps({ props: inProps, name: "CometAdminMyComponent" });
+    return (
+        // ...
+    )
+}
+```
+
+## `slotProps` and `restProps`
+
+To allow overriding the props of each slot, the `slotProps` object should be spread into each slot.
+Additionally, the `sx` and `className` props should be passed into the root slot, for example with `...restProps`.
+
+```tsx
+export function MyComponent(inProps: MyComponentProps) {
+    const { slotProps, ...restProps } = useThemeProps({
+        props: inProps,
+        name: "CometAdminMyComponent",
+    });
+    return (
+        <Root {...slotProps?.root} {...restProps}>
+            <Title {...slotProps?.title} />
+        </Root>
+    );
+}
+```
+
+## Owner state
+
+The `ownerState` is an object that includes all values that should impact the component's styling.
+It is passed into every slot that requires any of these values for its styling.
+
+```tsx
+type OwnerState = {
+    highlighted: boolean;
+};
+
+const Root = createComponentSlot("div")<MyComponentClassKey, OwnerState>({
+    componentName: "MyComponent",
+    slotName: "root",
+})(
+    ({ theme, ownerState }) => css`
+        ${ownerState.highlighted &&
+        css`
+            background-color: ${theme.palette.primary.main};
+        `}
+    `,
+);
+
+export function MyComponent(inProps: MyComponentProps) {
+    // ...
+    const ownerState: OwnerState = {
+        highlighted: getHighlightedState(),
+    };
+
+    return <Root ownerState={ownerState} {...slotProps?.root} {...restProps} />;
+}
+```
+
+## A slots `classesResolver`
+
+By default, a slots class name consists of the component and slot names, prefixed with `CometAdmin`.
+For example, the `root` slot of `MyComponent` would have the class name `CometAdminMyComponent-root`, while the `title` slot would have the class name `CometAdminMyComponent-title`.
+
+Additional class names can be added to a slot by returning an array of class keys in `classesResolver()`.
+This allows easy customization using its class name in a selector outside the component.
+
+In the following example, a component has a conditional `highlighted` state.
+If `true`, the class name `CometAdminMyComponent-highlighted` is added to the `root` slot by returning the `highlighted` class key in `classesResolver`.
+If `false`, `classesResolver` returns an empty array, so no additional class name is added.
```

Comment:
This isn't true. The array isn't empty, but `[false]` instead. Could this lead to problems?

<img width="319" alt="image" src="https://github.com/vivid-planet/comet/assets/48853629/e8cea44e-027f-4baf-8ca3-d0e12850276f">

Reply:
Good point, will re-write this. 

> Could this lead to problems?

No, `falsy` values are filtered out:

https://github.com/vivid-planet/comet/blob/89092f9e2b983cc689d6d2238d46ec4930992aa9/packages/admin/admin/src/helpers/createComponentSlot.tsx#L38